### ISSUE TITLE
Socket path corrupt?

### ISSUE THREAD

**Issue by ileitch** [IBM-Swift/BlueSocket#46]
When listening on a socket which includes a directory in the path, the socket name appears to be corrupted/truncated/something.

For example:
```swift
do {
    let listenSocket = try Socket.create(family: .unix)
    try listenSocket?.listen(on: "/tmp/ttipc.socket")
} catch {
   // ...
}
```

Results in the file:
```
➜  /tmp ls -al
srwxr-xr-x   1 ian   wheel    0 Dec 16 08:55 ttipc.soc?
```

Printing the directory contents in Swift yields the string: `"ttipc.soc\u{01}"`

Ultimately the client is unable to connect because it cannot find the socket.

However, if I first change to the `/tmp` directory before listening (using `FileManager.default.changeCurrentDirectoryPath("/tmp")` the socket is created as expected.

System:
* iPhone 7 Simulator (10.2)
* Xcode 8.2
* Apple Swift version 3.0.2 (swiftlang-800.0.63 clang-800.0.42.1)
* macOS 10.12.2

**Comment by billabt** [comment]
I'm on vacation until Jan 2.  If I have time before then I'll try and take a look. Otherwise, it'll be first on my list.

**Comment by billabt** [comment]
Fixed in 0.12.17.
